lib/logstash/inputs/s3.rb

```diff
@@ -113,6 +117,8 @@ def run(queue)
 
   public
   def list_new_files
+    stop unless @watch_for_new_files
```

Comment:
Sending the `stop` method while listing the files (and before processing them) will cause the processing phase to break iteration before actually processing the objects we found.

I think that this would go best inside the interval in the interval block in the `run` method, immediately after we send `process_files(queue)`.